tests/device_tests/test_msg_authorize_coinjoin.py

```diff
@@ -0,0 +1,391 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2020 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import btc, messages
+from trezorlib.exceptions import TrezorFailure
+from trezorlib.tools import parse_path
+
+from ..tx_cache import TxCache
+from .signtx import request_finished, request_input, request_meta, request_output
+
+B = messages.ButtonRequestType
+
+TX_CACHE_TESTNET = TxCache("Testnet")
+TX_CACHE_MAINNET = TxCache("Bitcoin")
+
+TXHASH_e5b7e2 = bytes.fromhex(
+    "e5b7e21b5ba720e81efd6bfa9f854ababdcddc75a43bfa60bf0fe069cfd1bb8a"
+)
+TXHASH_65b811 = bytes.fromhex(
+    "65b811d3eca0fe6915d9f2d77c86c5a7f19bf66b1b1253c2c51cb4ae5f0c017b"
+)
+
+PIN = "1234"
+ROUND_ID_LEN = 32
+
+pytestmark = pytest.mark.skip_t1
+
+
+@pytest.mark.setup_client(pin=PIN)
+def test_sign_tx(client):
+    with client:
+        client.use_pin_sequence([PIN])
+        btc.authorize_coinjoin(
+            client,
+            coordinator="www.example.com",
+            max_total_fee=10010,
+            fee_per_anonymity=5000000,  # 0.005 %
+            n=parse_path("m/84'/1'/0'"),
+            coin_name="Testnet",
+            script_type=messages.InputScriptType.SPENDWITNESS,
+        )
+
+    client.call_raw(messages.LockDevice())
```

Comment:
OK, what's the difference BTW? I got it from `clear_session()`.

Reply:
https://github.com/trezor/trezor-firmware/pull/1127/commits/c6f5aa92f78f268a159e529386ae2a5aff132c41

Reply:
`call_raw` will not do additional processing of Button/Pin/Passphrase requests and will not raise on received Failure

So in general it should only be used when we explicitly want to handle one of these things manually.
I think the usage in `clear_session` is to avoid the exception, and it also looks like it's wrong.